features/support/files_helpers.rb

```diff
@@ -23,6 +23,32 @@ def files_in_branches
 end
 
 
+def create_uncommitted_file options = {}
+  options.reverse_merge! default_uncommitted_file_attributes
+  IO.write options[:file_name], options[:file_content]
+  options[:file_name]
+end
+
+
+def default_uncommitted_file_attributes
+  {
+    file_name: 'uncommitted_file',
+    file_content: 'default uncommitted content'
+  }
+end
+
+
+def verify_uncommitted_file options
+  if options.key? :file_name
```

Comment:
Looking at this again, I realize that this `if` clause is unnecessary and can be removed.

* you reverse-merge a file_name into the options hash on the next line anyways if one is missing
* manually checking that certain parameters have certain keys is manually implementing a static type system. Ruby as a dynamically typed language shouldn't need to do this. If we wanted type checking, we would use a statically typed language. 
* if you really wanted to keep this check in, I suggest a guard clause:

  ```ruby
def verify_uncommitted_file options
  return fail 'Cannot verify uncommitted file without file_name'  unless options.key? :file_name
  options.reverse_merge! default_uncommitted_file_attributes
  ...
```

So I would try to take this if-clause out and see if it still works. I think it will. 

```ruby
def verify_uncommitted_file options
  options.reverse_merge! default_uncommitted_file_attributes
  expect(uncommitted_files).to include options[:file_name]
  expect(IO.read options[:file_name]).to eql options[:file_content]
end
```